### ISSUE TITLE
Vergleich einiger Excel-Funktionen hinzufügen

### ISSUE THREAD

**Issue by MassiveHiggsField** [omigos/ExcelCompare#1]
Folgende Funktionen sollten verglichen werden:

- [x] Title

       $this->getJWorkbook()->getProperties()->getCoreProperties()->setTitle($title);
    
- [ ] Autofilter

       $this->getJSheet()->setAutoFilter(
           new \java(
               'org.apache.poi.ss.util.CellRangeAddress',
               (int)$cellRange['firstRow'] - 1,
               (int)$cellRange['lastRow'] - 1,
               (int)$cellRange['firstCol'],
               (int)$cellRange['lastCol']
           )
       );
            
- [x] ColWidth

        $this->getJSheet()->setColumnWidth(
            $this->convertColumnLettersToNum($colAddress),
            (int)$width
        );

- [ ] ColGroup

        $this->getJSheet()->groupColumn(
            $this->convertColumnLettersToNum($fromAddress['col']) + 1,
            $this->convertColumnLettersToNum($toAddress['col'])
        );

- [x] FreezePane

        $this->getJSheet()->createFreezePane(
            (int)$this->convertColumnLettersToNum($cellAddress['col']),
            (int)$cellAddress['row']
        );

- [x] MergeCell

        $region = new \java(
            'org.apache.poi.ss.util.CellRangeAddress',
            (int)$cellRange['firstRow'] - 1,
            (int)$cellRange['lastRow'] - 1,
            (int)$cellRange['firstCol'],
            (int)$cellRange['lastCol']
        );

        $this->getJSheet()->addMergedRegion($region);

- [x] RowGroup

        $this->getJSheet()->groupRow(
            $fromAddress['row'],
            $toAddress['row'] - 1
        );

**Comment by KoesterD** [comment]
Ich habe nur für ColWidth und MergeCell einen Getter gefunden. Die Setter für den Rest finde ich, aber keine Möglichkeit das abzurufen. Mal sehen, ob ich noch was finde.

**Comment by MassiveHiggsField** [comment]
Also der Title sollte auch nen Getter haben, den benutzen wir auch:

    $this->getJWorkbook()->getProperties()->getCoreProperties()->getTitle();

**Comment by KoesterD** [comment]
Ja, stimmt. An RowGroup komme ich z.B. auch noch so dran (über getRow(row).getCTRow().getOutlineLevel()). Aber für z.B. FreezePane müsste ich POI forken :/

**Comment by MassiveHiggsField** [comment]
Ja mach erstmal soweit du ohne größeren Aufwand kommst, hake die Sachen oben ab und alles was noch fehlt, da sprechen wir dann nochma kurz.

**Comment by MassiveHiggsField** [comment]
Vlt hilft das hier schon eimal weiter: 

https://stackoverflow.com/questions/23004545/how-to-get-information-of-freeze-or-split-columns-in-excel-while-reading-using-a

**Comment by MassiveHiggsField** [comment]
Das hier hilft vlt noch mit den MergeCells:

https://stackoverflow.com/questions/18529467/find-merged-cell-in-excel-split-the-cells-and-write-those-in-new-spreadsheet

**Comment by KoesterD** [comment]
Wir wollen statt des Workbook-Titles den Sheet-Name vergleichen.

**Comment by KoesterD** [comment]
Hab jetzt noch so ein paar Einzelheiten gefunden und oben abgehakt. Ich bastel das jetzt noch zusammen und stelle dir dann eine Binary zum Testen bereit.

**Comment by KoesterD** [comment]
https://github.com/omigos/ExcelCompare/releases/tag/0.6.1-style2
In dem Release ist wieder eine Binary drin.

**Comment by MassiveHiggsField** [comment]
Also folgende Funktionen habe ich erfolgreich getestet:

- Title
- ColWidth
- FreezePane
- MergeCell

Die RowGroups konnte ich noch nicht testen, da die scheinbar zur Zeit nicht richtig funktionieren. Bezgl. der beiden übrig gebliebenen (ColGroup und Autofilter), gibt es da noch Hoffnung oder müssten wir wenn überhaupt eine XML-Diff Option einbauen oder so?

**Comment by KoesterD** [comment]
Was meinst du bzgl. RowGroups? Im Projekt ist ein Test, der funktioniert. Hast du Testdaten, wo das nicht mit funktioniert?

Bzgl. ColGroup und AutoFilter müsste man sich auf jeden Fall was cleveres einfallen lassen. XML-Diff oder POI patchen o.ä.

**Comment by MassiveHiggsField** [comment]
Sorry, sie funktionieren im Reporting nicht richtig, der Vergleich ist bestimmt spitze, kann den nur nicht prüfen ;)

**Comment by MassiveHiggsField** [comment]
Noch ein Nachtrag zu den RowGroups. @KoesterD kannst du hier noch einbauen, das er auch vergleicht ob die RowGroups collapsed oder nicht sind? Bei unseren Tests in https://github.com/OPMtechnology/SOD-Reporting/issues/552 ist uns aufgefallen, das er das nicht prüft.
